assets/js/googlesitekit/notifications/register-defaults.js

```diff
@@ -60,14 +60,23 @@ export function registerDefaults( notificationsAPI ) {
 			VIEW_CONTEXT_ENTITY_DASHBOARD_VIEW_ONLY,
 			VIEW_CONTEXT_SETTINGS,
 		],
-		checkRequirements: ( { select } ) => {
+		checkRequirements: async ( { select, resolveSelect } ) => {
+			// The getSetupErrorMessage selector relies on the resolution
+			// of the getSiteInfo() resolver.
+			await resolveSelect( CORE_SITE ).getSiteInfo();
```

Comment:
Thanks, @jimmymadon. All looks good, but let's combine all resolveSelect-s into one `await Promise.all()` call to make sure that all resolvers processed simultaneously instead of sequentially. This will reduce the overall loading time for notification requirement checks.

Reply:
Thanks @eugene-manuilov! I've combined the 2 simple obvious cases. Others can be combined too but we'd have to store the results into an array and the code will get a bit harder to read - so avoiding that for now.

Reply:
Not necessarily. We can call `await Promise.all( [ ..., resolveSelect( ... ).getRecoverableModules() ] )` without storing results and then call the `getRecoverableModules` selector when needed to get results.